mat64/dense.go

```diff
@@ -164,13 +175,53 @@ func (m *Dense) View(i, j, r, c int) Matrix {
 	t.mat.Data = t.mat.Data[i*t.mat.Stride+j : (i+r-1)*t.mat.Stride+(j+c)]
 	t.mat.Rows = r
 	t.mat.Cols = c
+	t.capRows -= i
+	t.capCols -= j
 	return &t
 }
 
+func (m *Dense) Grow(r, c int) {
```

Comment:
Should this return a matrix like view (and append)?
````
func (m *Dense) Grow(r, c int) Matrix
````
It is somewhat complicated to see the interactions with this and other functions, but they're roughly the same complexity as append, and the common case of  m = m.Grow(r,c) works okay.